src/gui/threejsScene.js

```diff
@@ -848,6 +866,38 @@ import { ViewFrustum, frustumVertexShader, frustumFragmentShader, MAX_VIEW_FRUST
         };
     };
 
+    // source: https://stackoverflow.com/questions/29758233/three-js-check-if-object-is-still-in-view-of-the-camera
+    exports.isObjectOnScreen = function(objectPosition) {
+        let frustum = new THREE.Frustum();
+        let matrix = new THREE.Matrix4();
+        matrix.multiplyMatrices(camera.projectionMatrix, camera.matrixWorldInverse);
+        frustum.setFromProjectionMatrix(matrix);
+        if (frustum.containsPoint(objectPosition)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // gets the position relative to groundplane (common coord system for threejsScene)
+    exports.getToolPosition = function(toolId) {
+        let toolSceneNode = realityEditor.sceneGraph.getSceneNodeById(toolId);
+        let groundPlaneNode = realityEditor.sceneGraph.getGroundPlaneNode();
+        // console.log('%c debugging tool position', 'color: orange');
+        // console.log(realityEditor.sceneGraph.convertToNewCoordSystem({x: 0, y: 0, z: 0}, toolSceneNode, groundPlaneNode));
+        return realityEditor.sceneGraph.convertToNewCoordSystem({x: 0, y: 0, z: 0}, toolSceneNode, groundPlaneNode);
+    }
+
+    // gets the direction the tool is facing, within the coordinate system of the groundplane
+    exports.getToolDirection = function(toolId) {
```

Comment:
This function and the one above are very useful, thanks for adding them!

Reply:
Ben originally added them in the spatial cursor actually. I just moved them into threejsScene.js.